test/action_controller/serialization_test.rb

```diff
@@ -420,6 +427,14 @@ def use_adapter?
           controller.get_serializer(Profile.new)
         })
       end
+
+      def test_render_object_xml
+        get :render_object_xml
+
+        hash = Hash.from_xml(@response.body).each_value.first
```

Comment:
So, ActiveModel::Serializers::XML does this.. https://github.com/rails/rails/blob/4-2-stable/activemodel/lib/active_model/serializers/xml.rb#L232-L234